packages/flutter/lib/src/material/input.dart

```diff
@@ -47,6 +47,7 @@ class InputField extends StatefulWidget {
     this.keyboardType: TextInputType.text,
     this.hintText,
     this.style,
+    this.hintStyle,
```

Comment:
The hintStyle/labelStyle seem highly opinionated for the InputContainer (which is used as a wrapper for the Input widget as well).  They are tied to the theme and the Dark/Lightness of the theme. 

After discussing offline, it seems best to keep the hintStyle override to the InputField, which is the non-material, open-ended widget for text input.

Tests added :)